docs/sdk-configuration.md

```diff
@@ -0,0 +1,492 @@
+# Design for configuring the SDK
+
+This document outlines some of the goals we have for user configuration of the SDK. It is a
+continuation of discussion started with https://github.com/open-telemetry/opentelemetry-java/issues/2022.
+
+## Target audiences
+
+There are a few different target audiences that are related to our configuration story.
+
+- Application developers, aka end-users. Often have no knowledge of tracing but want to add 
+OpenTelemetry to their app and see traces show up in a console. Application developers will increase
+in number forever, while the below are more constant.
+
+- Dev-ops / framework developers. Write components or frameworks to support providing tracing to
+their application developers. May write custom SDK extensions such as exporters, samplers, to fit
+with their internal infrastructure and as such have some familiarity with tracing at least as the
+SDK presents it.
+
```

Comment:
I think we bundle two types here: authors of custom exporters/samplers and frameworks authors. The latter will handle configuration, but the former will just write a component according to an interface.

Reply:
the framework authors might need to do both (see Sleuth), but your point is still a good one.